jdaviz/core/unit_conversion_utils.py

```diff
@@ -53,6 +55,62 @@ def all_flux_unit_conversion_equivs(pixar_sr=None, cube_wave=None):
     return equivs
 
 
+def viewer_flux_conversion_equivalencies(values, spec):
+    """
+    Generate a list of flux and surface brightness unit conversion equivalencies
+    specifically for converting units in the viewer, accounting for the special
+    case of viewer limits that may need to index the spectral axis or list of
+    pixel scale factors.
+
+    This function assumes that if exactly two values are being converted, they
+    represent the y-axis limits of the viewer. In this case, the first spectral
+    axis value is used for spectral density conversions, and the min/max value
+    of pixel scale factor are used if there are more than 2 values in
+    meta['_pixel_scale_factor'].
+
+    Parameters:
+    ----------
+    values : array-like
+        The values to be converted, which may represent flux or surface brightness.
+    spec : Spectrum1D
+
+    Returns:
+    -------
+    equivs : list
+        A list of unit equivalencies for flux and surface brightness conversions.
+    """
+
+    # if we are converting only 2 values, assume it is a viewer y limits case.
+    is_viewer_limits = len(values) == 2
+
+    # for viewer limits case, use only the 0th spectral axis value for spectral_density
+    spectral_values = spec.spectral_axis
+    if not np.isscalar(values) and is_viewer_limits:
+        spectral_values = spectral_values[0]
+
+    # Need this for setting the y-limits but values from viewer might be downscaled
+    if len(values) != spectral_values.size:
+        spectral_values = spec.spectral_axis[0]
+
+    # Next, pixel scale factor
+    pix_fac = None
+    if '_pixel_scale_factor' in spec.meta:
+        pix_fac = spec.meta['_pixel_scale_factor']
+
+        if isinstance(pix_fac, u.Quantity):
+            pix_fac = pix_fac.value
```

Comment:
any way this could be in some other units (should we use `tovalue` instead)?

Reply:
I believe we assume in other places that this is always in units of steradian